airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java

```diff
@@ -128,7 +128,7 @@ protected void acceptTracked(AirbyteMessage message) throws Exception {
     if (message.getType() == Type.RECORD) {
       final AirbyteRecordMessage recordMessage = message.getRecord();
       final AirbyteStreamNameNamespacePair stream = AirbyteStreamNameNamespacePair.fromRecordMessage(recordMessage);
-      final String data = Jsons.serialize(message);
+      final String data = Jsons.serialize(message.getRecord().getData());
```

Comment:
Would we be able to avoid most of these problems if the record validator was for an `AirbyteRecordMessage` instead of a String?

Then the serialization would live within the record validator and could be easily tested.

Reply:
not a bad idea. I'll create a ticket for it.